### ISSUE TITLE
loadImageWithURL:使用疑惑

### ISSUE THREAD

**Issue by KeymonWong** [rs/SDWebImage#2170]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

```objective-c
    for (NSString *urlPath in aDatasource) {
        [[SDWebImageManager sharedManager] loadImageWithURL:[NSURL URLWithString:urlPath] options:SDWebImageRefreshCached|SDWebImageRetryFailed progress:^(NSInteger receivedSize, NSInteger expectedSize, NSURL * _Nullable targetURL) {

        } completed:^(UIImage * _Nullable image, NSData * _Nullable data, NSError * _Nullable error, SDImageCacheType cacheType, BOOL finished, NSURL * _Nullable imageURL) {
            if (image && finished) {
                [imageArray addObject:image];
            }
        }];
    }
```

请问这样一段代码，循环下载一组图片，这样合理吗？还有completed是一张图片下载完成之后回调？还是所有图片下载完成之后再回调？使用loadImageWithURL:如何判断所有下载任务都进行完了？我这里使用了dispatch_group监听所有任务是否下载完成，但是回到主线程加载图片的时候不显示

**Comment by dreampiggy** [comment]
Use `-[SDWebImagePrefetcher prefetchURLs:progress:completed:]` instead is the better choice.

Use `dispatch_group` seems will work, but `SDWebImageRefreshCached` this options is a little misunderstanding. It will make your completion block called 1 or 2 times. The first one is from `SDImageCache`, the second one may called if the downloaded image data not equal to cached data. So this is not recommend to do this :)
